### PR TITLE
Ref: Hint is passed across Sentry static class, Hub and Client

### PR DESCRIPTION
**Pull request by marandaneto**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [X] Refactoring

## :scroll: Description
Ref: Hint is passed across Sentry static class, Hub and Client

## :bulb: Motivation and Context
EventProcessors support Hint, in order to support that, Sentry static class, hub and Client should be able to get it

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

File: dart/lib/src/client.dart [hunk 1]
```diff
@@ -99,6 +99,7 @@ abstract class SentryClient {
     SentryEvent event, {
     StackFrameFilter stackFrameFilter,
```
Comment (marandaneto) [inline dart/lib/src/client.dart:100]: https://github.com/getsentry/sentry-dart/projects/1#card-47872158

File: dart/lib/src/noop_client.dart [hunk 2]
```diff
@@ -34,21 +35,30 @@ class NoOpSentryClient implements SentryClient {
...
-    String template,
-    List<dynamic> params,
-    Scope scope,
-  }) =>
+  Future<SentryId> captureMessage(String message,
+          {SentryLevel level = SentryLevel.info,
+          String template,
+          List<dynamic> params,
+          Scope scope,
+          dynamic hint}) =>
```
Comment (rxlabz) [inline dart/lib/src/noop_client.dart]: please add a trailing comma :)

File: dart/lib/src/noop_hub.dart [hunk 3]
```diff
@@ -19,20 +19,23 @@ class NoOpHub implements Hub {
...
-    String message, {
-    SentryLevel level = SentryLevel.info,
-    String template,
-    List params,
-  }) =>
+  Future<SentryId> captureMessage(String message,
+          {SentryLevel level = SentryLevel.info,
+          String template,
+          List params,
+          dynamic hint}) =>
```
Comment (rxlabz) [inline dart/lib/src/noop_hub.dart]: please add a trailing comma :)

File: dart/lib/src/sentry.dart [hunk 4]
```diff
@@ -45,30 +45,28 @@ class Sentry {
   }
 
   /// Reports an [event] to Sentry.io.
-  static Future<SentryId> captureEvent(SentryEvent event) async {
-    return currentHub.captureEvent(event);
+  static Future<SentryId> captureEvent(SentryEvent event,
+      {dynamic hint}) async {
```
Comment (rxlabz) [inline dart/lib/src/sentry.dart]: please add a trailing comma :)

File: dart/lib/src/sentry.dart [hunk 5]
```diff
@@ -45,30 +45,28 @@ class Sentry {
...
-      message,
-      level: level,
-      template: template,
-      params: params,
-    );
+  Future<SentryId> captureMessage(String message,
+      {SentryLevel level,
+      String template,
+      List<dynamic> params,
+      dynamic hint}) async {
```
Comment (rxlabz) [inline dart/lib/src/sentry.dart]: please add a trailing comma :)

File: dart/lib/src/sentry.dart (hunk 4 above)
Comment (marandaneto) [reply dart/lib/src/sentry.dart]: thx, looks like `dartfmt -n --set-exit-if-changed ./` is not clever enough

File: dart/lib/src/sentry.dart [hunk 6]
```diff
@@ -45,30 +45,28 @@ class Sentry {
...
+    return currentHub.captureMessage(message,
+        level: level, template: template, params: params, hint: hint);
```
Comment (rxlabz) [inline dart/lib/src/sentry.dart]: please add a trailing comma :)

File: dart/lib/src/sentry.dart [hunk 7]
```diff
@@ -45,30 +45,28 @@ class Sentry {
...
 
   /// Reports the [exception] and optionally its [stackTrace] to Sentry.io.
   static Future<SentryId> captureException(
     dynamic error, {
     dynamic stackTrace,
+    dynamic hint,
   }) async {
-    return currentHub.captureException(error, stackTrace: stackTrace);
+    return currentHub.captureException(error,
+        stackTrace: stackTrace, hint: hint);
```
Comment (rxlabz) [inline dart/lib/src/sentry.dart]: please add a trailing comma :)

**Review by rxlabz — APPROVED** [review]
LGTM (beside missing trailing comma :) )

**Pull request merged**